mainwindow.cpp

```diff
@@ -270,6 +276,86 @@ namespace dealii
 
 
 
+    void MainWindow::show_settings ()
+    {
+      settings_dialog = new SettingsDialog(gui_settings,this);
+
+      connect(settings_dialog, SIGNAL(accepted()), this, SLOT(apply_settings()));
+      settings_dialog->exec();
+      disconnect(settings_dialog, SIGNAL(accepted()), this, SLOT(apply_settings()));
+    }
+
+
+
+    void MainWindow::apply_settings ()
+    {
+      update_visible_items();
+      update_font();
+    }
+
+
+
+    void MainWindow::toggle_visible_default_items()
+    {
+      const bool hide_default_values = gui_settings->value("Settings/hideDefault", false).toBool();
+      gui_settings->setValue("Settings/hideDefault", !hide_default_values);
+      update_visible_items();
+    }
+
+
+
+    void MainWindow::update_visible_items()
+    {
+      const bool hide_default_values = gui_settings->value("Settings/hideDefault", false).toBool();
+
+      if (hide_default_values)
+        {
+        for (int i = 0; i < tree_widget->topLevelItemCount(); ++i)
+            hide_default_item(tree_widget->topLevelItem(i));
+        hide_default->setChecked(true);
+        }
+      else
+        {
+          QTreeWidgetItemIterator it(tree_widget,QTreeWidgetItemIterator::Hidden);
+          while (*it)
+            {
+              (*it)->setHidden(false);
+              ++it;
+            }
+          hide_default->setChecked(false);
+          }
+    }
+
+
+
+    bool MainWindow::hide_default_item(QTreeWidgetItem *item)
```

Comment:
I think the name is misleading. You are hiding items whose values equal the default. Maybe `hide_items_with_default_value`()?